compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -311,29 +311,19 @@ impl Visibility {
 pub enum BoundConstness {
     /// `T: Trait`
     NotConst,
+    /// `T: const Trait`
+    Const,
     /// `T: ~const Trait`
     ///
     /// Requires resolving to const only when we are in a const context.
     ConstIfConst,
 }
 
-impl BoundConstness {
-    /// Reduce `self` and `constness` to two possible combined states instead of four.
-    pub fn and(&mut self, constness: hir::Constness) -> hir::Constness {
```

Comment:
This was unused even before this PR. Well, not sure if any tools use it after all. Let's see what CI has to say.

Reply:
yeah, this should be removed